Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1968,8 +1962,7 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
         defer { deferredEventEmittingBlock?() }
```

Comment:
[nit] with this new locking scheme, we don't need the deferred event emitting block. Since we're not deferring to unlock the lock, we can just trigger listening and emit the event at the end of the function outside of the `lock.write` block